llvm/lib/Transforms/Utils/PromoteMemoryToRegister.cpp

```diff
@@ -1112,6 +1113,12 @@ void PromoteMem2Reg::RenamePass(BasicBlock *BB, BasicBlock *Pred,
         for (unsigned i = 0; i != NumEdges; ++i)
           APN->addIncoming(IncomingVals[AllocaNo], Pred);
 
+        if (APN->isComplete() &&
+            APN->getFunction()->hasFnAttribute("no-signed-zeros-fp-math") &&
```

Comment:
Unfortunately these attributes have explicit boolean parameters. You should use .getValueAsBool() on this. You should add a negative test with an explicit ="false" value